### ISSUE TITLE
SuRE-seq: Initial Deep Models for MPRA Regression and Classification

### ISSUE THREAD

**Issue by rmovva** [kundajelab/mpra#2]
**26 June 2017**

My first deep learning models finished training today! For the first pass, I trained a regression model using the 2.2M fragments that were positive across both of the SuRE replicates (Spearman corr = 0.34). I didn't do any curation to select a subsample of these 2.2M fragments, since (1) I ran IDR on this set, and as noted [here](https://github.com/kundajelab/mpra/issues/1#issuecomment-310828047), there were no clear "signal" and "noise" clusters within this group (that is - the fragments that were most reproducible according to IDR did not have a better Spearman correlation than the rest of the fragments). And (2), using fewer than 2M fragments would probably worsen performance given the diversity of the sequences (basically sampled i.i.d. from the genome).

The 2.2M fragments were split into 50K validation (chr18), 100K test (chr8), and 2M training (chrs1-22 except chr18 and chr8). I've left out chrX and chrY for now, for no particular reason; I could probably use them. Since most of the sequences were fewer than 2000bp, but the maximum size was 2000bp (a small number of fragments were >2kb, but I left them out), I centered the sequences and zero-padded the sides. The model has a Basset-like architecture (model summary is below) and is multitasked on average signal, replicate 1 signal, and replicate 2 signal.

I trained with a epoch size of 200K, batch size 500, and early stopping when performance hasn't improved for 5 epochs. Over a few runs, the model generally takes 6-7 optimize and then plateaus. I used Spearman correlation between predictions and signals as the primary metric. Validation performance, averaged across the three tasks, was Spearman corr = ~0.23 (training performance was 0.314). For the individual tasks, validation scores were 0.256, 0.126, and 0.301 for average signal, rep1 signal, and rep2 signal respectively. 

@akundaje Here are some questions/concerns I have based on these initial results:

1. Can a better architecture improve performance nontrivially? Is a hyperparameter search worth doing? 
2. Among the three tasks, replicate 1 signal is by far the hardest to predict, with prediction Spearman correlation maxing out at approx 0.12 to 0.13. Replicate 2 signal is ~0.30. Would it make sense to just try and optimize performance on replicate 2, and remove replicate 1 from further analysis? Replicate 1 also has 96.5% zero signal vs. 85% zero signal in replicate 2, so it seems like replicate 1 went poorly for some reason.
3. Given that I'm already getting 0.23 correlation with average and 0.30 correlation with replicate 2, and corr(rep1, rep2) = 0.34, how much more room is there to go with the model? If 0.34 is the biological upper bound, I imagine there's some things that the model cannot learn purely from sequence features.
4. I think we should be more concerned about the quality of this data, especially for interpretation. I was talking to @jisraeli earlier today about interpreting models, and he showed me one of his simulations where Deeplift couldn't identify the true biological motif feature until the model had achieved AUROC 0.994 (there were still confounding factors at AUROC = 0.98). Given that we're going to see, at best, Spearman = 0.3-0.4 with this model, is it even worth spending time trying to interpret these models? Should I move to another dataset?
5. Sequence strandedness: some of the fragments that were assayed were from the negative strand - I think this means that they were in the opposite direction when put in the plasmid. I'm using these sequences in the training set, but I think most of the negative strand fragments had lower signal – since most promoter-driving regulatory grammars are on the positive strand, AFAIK. Also, I think negative stranded fragments may be less reproducible, and treating them as separate also prevents me from doing reverse complement augmentation. So any insight on how to handle strandedness would be helpful.

Current model architecture:
![screen shot 2017-06-27 at 3 13 59 pm](https://user-images.githubusercontent.com/17256372/27612438-56d814f0-5b4b-11e7-8c1d-21e8ae8272d8.png)
